Issue saving world.

```java
2020-11-17 20:09:41.904 [main] INFO  - [RealTriassic: Saving...]
2020-11-17 20:09:42.050 [main] FATAL - Unhandled exception executing command "save-all" in save-all: cn.nukkit.utils.ChunkException: Error saving chunk (105, 21)
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:243)
	at cn.nukkit.level.format.generic.BaseLevelProvider.saveChunks(BaseLevelProvider.java:306)
	at cn.nukkit.level.Level.saveChunks(Level.java:1318)
	at cn.nukkit.level.Level.save(Level.java:1309)
	at cn.nukkit.command.defaults.SaveCommand.execute(SaveCommand.java:34)
	at cn.nukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:258)
	at cn.nukkit.Server.dispatchCommand(Server.java:962)
	at cn.nukkit.Player.handleDataPacket(Player.java:3060)
	at cz.creeperface.nukkit.gac.player.NukkitCheatPlayer.handleDataPacket(NukkitCheatPlayer.kt:74)
	at cn.nukkit.network.Network.lambda$processPackets$3(Network.java:285)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at cn.nukkit.network.Network.processPackets(Network.java:283)
	at cn.nukkit.network.Network.processBatch(Network.java:269)
	at cn.nukkit.Player.handleDataPacket(Player.java:2303)
	at cz.creeperface.nukkit.gac.player.NukkitCheatPlayer.handleDataPacket(NukkitCheatPlayer.kt:74)
	at cn.nukkit.network.RakNetInterface.process(RakNetInterface.java:75)
	at cz.creeperface.nukkit.gac.V.process(GACNetworkInterface.kt)
	at cn.nukkit.network.Network.processInterfaces(Network.java:158)
	at cn.nukkit.Server.tick(Server.java:1361)
	at cn.nukkit.Server.tickProcessor(Server.java:1144)
	at cn.nukkit.Server.start(Server.java:1104)
	at cn.nukkit.Server.<init>(Server.java:760)
	at cn.nukkit.Nukkit.main(Nukkit.java:120)
Caused by: java.io.IOException: Negative seek offset
	at java.io.RandomAccessFile.seek(RandomAccessFile.java:555)
	at cn.nukkit.level.format.anvil.RegionLoader.saveChunk(RegionLoader.java:120)
	at cn.nukkit.level.format.anvil.RegionLoader.writeChunk(RegionLoader.java:153)
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:241)
	... 22 more

2020-11-17 20:09:42.050 [main] ERROR - Throwing
cn.nukkit.utils.ChunkException: Error saving chunk (105, 21)
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:243) ~[server.jar:?]
	at cn.nukkit.level.format.generic.BaseLevelProvider.saveChunks(BaseLevelProvider.java:306) ~[server.jar:?]
	at cn.nukkit.level.Level.saveChunks(Level.java:1318) ~[server.jar:?]
	at cn.nukkit.level.Level.save(Level.java:1309) ~[server.jar:?]
	at cn.nukkit.command.defaults.SaveCommand.execute(SaveCommand.java:34) ~[server.jar:?]
	at cn.nukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:258) ~[server.jar:?]
	at cn.nukkit.Server.dispatchCommand(Server.java:962) ~[server.jar:?]
	at cn.nukkit.Player.handleDataPacket(Player.java:3060) ~[server.jar:?]
	at cz.creeperface.nukkit.gac.player.NukkitCheatPlayer.handleDataPacket(NukkitCheatPlayer.kt:74) ~[?:?]
	at cn.nukkit.network.Network.lambda$processPackets$3(Network.java:285) ~[server.jar:?]
	at java.util.ArrayList.forEach(ArrayList.java:1259) [?:1.8.0_262]
	at cn.nukkit.network.Network.processPackets(Network.java:283) [server.jar:?]
	at cn.nukkit.network.Network.processBatch(Network.java:269) [server.jar:?]
	at cn.nukkit.Player.handleDataPacket(Player.java:2303) [server.jar:?]
	at cz.creeperface.nukkit.gac.player.NukkitCheatPlayer.handleDataPacket(NukkitCheatPlayer.kt:74) [GAC-1.0.jar:?]
	at cn.nukkit.network.RakNetInterface.process(RakNetInterface.java:75) [server.jar:?]
	at cz.creeperface.nukkit.gac.V.process(GACNetworkInterface.kt) [GAC-1.0.jar:?]
	at cn.nukkit.network.Network.processInterfaces(Network.java:158) [server.jar:?]
	at cn.nukkit.Server.tick(Server.java:1361) [server.jar:?]
	at cn.nukkit.Server.tickProcessor(Server.java:1144) [server.jar:?]
	at cn.nukkit.Server.start(Server.java:1104) [server.jar:?]
	at cn.nukkit.Server.<init>(Server.java:760) [server.jar:?]
	at cn.nukkit.Nukkit.main(Nukkit.java:120) [server.jar:?]
Caused by: java.io.IOException: Negative seek offset
	at java.io.RandomAccessFile.seek(RandomAccessFile.java:555) ~[?:1.8.0_262]
	at cn.nukkit.level.format.anvil.RegionLoader.saveChunk(RegionLoader.java:120) ~[server.jar:?]
	at cn.nukkit.level.format.anvil.RegionLoader.writeChunk(RegionLoader.java:153) ~[server.jar:?]
	at cn.nukkit.level.format.anvil.Anvil.saveChunk(Anvil.java:241) ~[server.jar:?]
	... 22 more
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,7 @@ Click the link above to see the future.
 ### Fixes
 - _Alpha Bug_: Fixes entities from the MobPlugin going into the ground (compatibility issue)
 - [#770] Bamboo not dropping when broken, were also affecting blocks with id > 255
+- [#776] Error saving region files with content over 2 GB
 
 ## 1.3.2.0-PN-ALPHA.2 - 2020-11-21
 
@@ -623,3 +624,4 @@ Fixes several anvil issues.
 [#475]: https://github.com/PowerNukkit/PowerNukkit/issues/475
 [#544]: https://github.com/PowerNukkit/PowerNukkit/issues/544
 [#770]: https://github.com/PowerNukkit/PowerNukkit/issues/770
+[#776]: https://github.com/PowerNukkit/PowerNukkit/issues/776

--- a/src/main/java/cn/nukkit/level/format/anvil/RegionLoader.java
+++ b/src/main/java/cn/nukkit/level/format/anvil/RegionLoader.java
@@ -44,7 +44,7 @@ public class RegionLoader extends BaseRegionLoader {
         try {
         Integer[] table = this.locationTable.get(index);
         RandomAccessFile raf = this.getRandomAccessFile();
-        raf.seek(table[0] << 12);
+        raf.seek((long)table[0] << 12L);
         int length = raf.readInt();
         byte compression = raf.readByte();
         if (length <= 0 || length >= MAX_SECTOR_LENGTH) {
@@ -117,7 +117,7 @@ public class RegionLoader extends BaseRegionLoader {
 
         this.locationTable.put(index, table);
         RandomAccessFile raf = this.getRandomAccessFile();
-        raf.seek(table[0] << 12);
+        raf.seek((long)table[0] << 12L);
 
         BinaryStream stream = new BinaryStream();
         stream.put(Binary.writeInt(length));
@@ -172,7 +172,7 @@ public class RegionLoader extends BaseRegionLoader {
             if (table[0] == 0 || table[1] == 0) {
                 continue;
             }
-            raf.seek(table[0] << 12);
+            raf.seek((long)table[0] << 12L);
             byte[] chunk = new byte[table[1] << 12];
             raf.readFully(chunk);
             int length = Binary.readInt(Arrays.copyOfRange(chunk, 0, 3));
@@ -197,7 +197,7 @@ public class RegionLoader extends BaseRegionLoader {
                 this.lastSector += sectors;
                 this.locationTable.put(i, table);
             }
-            raf.seek(table[0] << 12);
+            raf.seek((long)table[0] << 12L);
             byte[] bytes = new byte[sectors << 12];
             ByteBuffer buffer1 = ByteBuffer.wrap(bytes);
             buffer1.put(chunk);
@@ -269,10 +269,10 @@ public class RegionLoader extends BaseRegionLoader {
                 shift += sector - lastSector - 1;
             }
             if (shift > 0) {
-                raf.seek(sector << 12);
+                raf.seek((long)sector << 12L);
                 byte[] old = new byte[4096];
                 raf.readFully(old);
-                raf.seek((sector - shift) << 12);
+                raf.seek((long)(sector - shift) << 12L);
                 raf.write(old);
             }
             Integer[] v = this.locationTable.get(index);

